src/pp/if_parser.rs

```diff
@@ -0,0 +1,436 @@
+use crate::token::{Integer, PreprocessorError, Punct};
+
+use super::{Define, Location, MELexer, MacroProcessor, Step, StepExit, Token, TokenValue};
+use std::{collections::HashMap, rc::Rc, vec::IntoIter};
+
+struct IfLexer<'macros> {
+    tokens: IntoIter<Token>,
```

Comment:
nit: maybe use vec::IntoIter, I got confused for a second about what that was doing.